Sources/Client/Util/Updater.swift

```diff
@@ -266,6 +266,39 @@ public final class Updater: ObservableObject {
       }
     }
   }
+  
+  /// Check if a commit (by its SHA) exists on a given branch.
+  ///
+  /// - Returns: If the commit exists on the branch
+  private static func doesCommitExistOnBranch(commit: String, branch: String) -> Bool {
+    let url = URL(string: "https://api.github.com/repos/stackotter/delta-client/compare/\(branch)...\(commit)")!
+    if let data = try? Data(contentsOf: url) {
+      let status = try? CustomJSONDecoder().decode(GitHubComparison.self, from: data).status
+      if status == "behind" || status == "identical" {
+        return true
+      }
+    }
+    return false
+  }
+  
+  /// If the current version is on the main branch, check if a newer commit is available.
+  ///
+  /// - Returns: Whether or not an "unstable" update is available from the main branch.
+  static func isUpdateAvailable() -> Bool {
+    if let currentVersionString = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String {
+      if let range = currentVersionString.range(of: "commit: ") {
+        // Check if this current version is from the main branch
+        if doesCommitExistOnBranch(commit: String(currentVersionString[range.upperBound...]), branch: "main") {
```

Comment:
I think it would be better to refactor this to be like `if getBranchComparisonStatus(commit: ..., branch: "main") == .behind {` because that would avoid the need for relying on the error handling on the next line and it would just clean up the code a bit more and make the `doesCommitExistOnBranch` function a bit more reusable